php/functions.php

```diff
@@ -47,30 +42,126 @@ function addParagraphs(array $paragraphs) :string {
+ */
+function editParagraphDropdown (array $paragraphs) : string {
     $paragraphList = '';
-    foreach ($editParagraphs as $paragraph){
-        $paragraphList .= '<option' . $paragraph['id'] . '>' . $paragraph['paragraph'] . '</option>'''
+    $i = 1 ;
+    foreach ($paragraphs as $paragraph){
+        if (is_string($paragraph['paragraph']) && array_key_exists('paragraph', $paragraph)) {
+            $charPreview = substr($paragraph['paragraph'], 0, 20);
+            $paragraphList .= '<option value=' . $paragraph['id'] . '>paragraph ' . $i++ . ' ' . $charPreview . '</option>';
+        } else {
+            $paragraphList .= '';
+        }
     }
     return $paragraphList;
 }
 
+/**
+ * this function retrieves the paragraph that is chosen in the dropdown list
+ *
+ * @param $db PDO calls the database from the db_query because this function requires it
+ *
+ * @param $editChoice string outputs a number of the id that will be used by the WHERE in MySQL
+ *
+ * @return mixed retrieves one row of the result by the SELECT in MySQL
+ */
+function getEdit(PDO $db, string $editChoice){
+    if ($editChoice == "choose"){
+        header('Location: admin_page.php');
+    } else {
+        $query = $db->prepare("SELECT `paragraph` FROM `about_me` WHERE `id` = :choiceId;");
+        $query->bindParam(':choiceId', $editChoice);
+        $query->execute();
+        return $query->fetch();
+    }
+}
+
+/**
+ * this function enables the html page to call and put the paragraph that is chosen from the dropdown and into the textarea for edit
+ *
+ * @param array $getEdit is the variable that retrieves the output of the getEdit function, which is the selected paragraph in the dropdown
+ *
+ * @return string returns a variable that can be assigned in the admin page and can be put into the textarea, the textarea will be populated by the chosen paragraph
+ */
+function pasteEdit(array $getEdit) : string {
+    $editPopulate = '';
+    if (is_string($getEdit['paragraph']) &&array_key_exists('paragraph', $getEdit)) {
+        $editPopulate .= $getEdit['paragraph'];
+    } else {
+        $editPopulate .= '';
+    }
+    return $editPopulate;
+}
 
+/**
+ * this function updates the existing paragraph in the database with the text that was submitted in the edit form text area
+ *
+ * @param PDO $db calls the database from the db_query because this function requires it
+ *
+ * @param $editChoice string outputs a number of the id that will be used by the WHERE in MySQL
+ *
+ * @param $newParagraph string is a variable that gets the text that was submitted in the 'edit paragraph' text area
+ *
+ * @return bool the function will only be called when the user press 'submit' and if true, will update the selected paragraph with the new paragraph
+ */
+function editParagraph(PDO $db, string $submitChoice, string $newParagraph) : bool {
+    if (empty(trim($newParagraph))){
+        header('Location: admin_page.php');
```

Comment:
Redirect shouldn't be here in your function methinks

Reply:
nowRichard thinks so too!